src/features/postSlice.js

```diff
@@ -0,0 +1,94 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+
+import axios from "axios";
+
+const initialState = {
+  posts: [],
+};
+
+export const addPost = createAsyncThunk(
+  "post/addPost",
+  async ({ postInput, token }, { rejectWithValue }) => {
+    try {
+      const response = await axios({
+        method: "post",
+        url: "api/posts",
+        headers: { authorization: token },
+        data: { postData: { content: postInput } },
+      });
+      return response.data;
+    } catch (err) {
+      rejectWithValue("something went wrong");
+    }
+  }
+);
+export const deletePost = createAsyncThunk(
+  "post/deletePost",
+  async ({ token, _id }, { rejectWithValue }) => {
+ 
+    try {
+      const response = await axios({
+        method: "delete",
+        url: `api/posts/${_id}`,
+        headers: { authorization: token },
+      });
+      return response.data;
+    } catch (err) {
+      console.log(err)
+      rejectWithValue("something went wrong");
+    }
+  }
+);
+export const editPost = createAsyncThunk(
+  "post/editPost",
+  async ({ postInput, token,_id }, { rejectWithValue }) => {
+    try {
+      const response = await axios({
+        method: "post",
+        url: `api/posts/edit/${_id}`,
+        headers: { authorization: token },
+        data: { postData: { content: postInput } },
+      });
```

Comment:
Learnt this new format of performing axios api calls. Looks good Aryan!! Great work!!